src/index.ts

```diff
@@ -48,6 +48,10 @@ export const localStorageMiddleware = (keys : string[], rehydrateState : boolean
     });
 
     return rehydrateState
-        ? [localStorageProvider, rehydrateApplicationState(keys)]
+        ? [localStorageProvider, rehydrateApplicationState(
+            rehydrateState.filter(state => {
```

Comment:
I would like this to be moved out into it's own method, something like `validateRehydrateKeys`, to be more explicit about the action being performed. Also, I think it would be nice to throw an error when a supplied key is not found in the `keys` list. I want to fail loudly in this case as 99% of the time it is a typo.